File: test_date.js
const dayMs = 24 * 60 * 60 * 1000;
let maxD = new Date();
let startWin = new Date(maxD.getTime() - 30 * dayMs);
let endWin = maxD;

const graphData = {};
let currentDay = new Date(startWin);
const limitDay = new Date(Math.min(endWin.getTime(), new Date().getTime()));
limitDay.setHours(23, 59, 59, 999);

let count = 0;
while (currentDay <= limitDay) {
    const dayKey = currentDay.getFullYear() + '-' + String(currentDay.getMonth() + 1).padStart(2, '0') + '-' + String(currentDay.getDate()).padStart(2, '0');
    graphData[dayKey] = { freeCalls: 0, paidCalls: 0 };
    currentDay.setDate(currentDay.getDate() + 1);
    count++;
}
console.log("Count:", count);
console.log("Keys:", Object.keys(graphData));

File: test_date.py
import datetime

maxD = datetime.datetime.now()
startWin = maxD - datetime.timedelta(days=30)
endWin = maxD

limitDay = min(endWin, datetime.datetime.now())
limitDay = limitDay.replace(hour=23, minute=59, second=59, microsecond=999000)

graphData = {}
currentDay = startWin

count = 0
while currentDay <= limitDay:
    dayKey = currentDay.strftime("%Y-%m-%d")
    graphData[dayKey] = {"freeCalls": 0, "paidCalls": 0}
    currentDay += datetime.timedelta(days=1)
    count += 1

print("Count:", count)
print("Keys:", list(graphData.keys()))
